L/libcifpp/build_tarballs.jl

```diff
@@ -0,0 +1,116 @@
+using BinaryBuilder, Pkg
+using BinaryBuilderBase: default_host_platform
+
+name = "libcifpp"
+version = v"5.1.0"
+
+# url = "https://github.com/PDB-REDO/libcifpp"
+# description = "Library containing code to manipulate mmCIF and PDB files"
+
+sources = [
+    GitSource("https://github.com/PDB-REDO/libcifpp",
+              "836aed6ea9a227b37e5b0d9cbcb1253f545d0778"),
+    DirectorySource("./bundled"),
+]
+
+# TODO
+# - the CCD (chemical components dictionary) normally gets downloaded
+#   by default and saved under share/libcifpp/components.cif, but we
+#   don't do this
+#
+#   This file is quite large (88MB gzipped, 380MB decompressed as of
+#   July 2023). Therefore, download and installation of this file is
+#   disabled for now, but not sure what functionality is impacted by
+#   this. We should maybe think of a better way of sharing
+#   components.cif between different packages that need it.
+#   Ref: https://www.wwpdb.org/data/ccd
+
+# Note: upstream seems to recommend gcc >= 10
+#       https://github.com/PDB-REDO/libcifpp/issues/39
+
+# Note: test suite (`make test`) is run if we are building for the
+# build host platform BinaryBuilderBase.default_host_platform, which
+# is the case for `$target == $MACHTYPE` inside the shell script.
+#
+# The tests only pass with the correct cxxabi (-cxx11), so we create a
+# MACHTYPE_FULL variable to pass to the shell script which can there
+# be matched against to bb_full_target.
+#
+# TODO: can we get the "libgfortran5" from default_host_platform?
+#
+# Convert x86_64-linux-musl-cxx11 -> x86_64-linux-musl-libgfortran5-cxx11
+const M = split(triplet(default_host_platform), "-")
+const MACHTYPE_FULL = join((M[1:3]..., "libgfortran5", M[4:end]...), "-")
+
+script =
+"""
+MACHTYPE_FULL=$MACHTYPE_FULL
+""" * raw"""
+cd $WORKSPACE/srcdir/libcifpp*/
+
+# mingw doesn't have ioctl
+# upstream PR: https://github.com/PDB-REDO/libcifpp/pull/45
+atomic_patch -p1 ../patches/mingw-no-ioctl.patch
+
+mkdir build && cd build
+
+CFG_TESTING="-DENABLE_TESTING=OFF"
+if [[ "${bb_full_target}" == "${MACHTYPE_FULL}" ]]; then
+    # build the tests if we are building for the build host platform
+    CFG_TESTING="-DENABLE_TESTING=ON"
+fi
+
+# use gcc/g++ on apple
+if [[ "${target}" == *-apple-darwin* ]]; then
+    CMAKE_TARGET_TOOLCHAIN="$(dirname "${CMAKE_TARGET_TOOLCHAIN}")/target_${target}_gcc.cmake"
```

Comment:
https://docs.binarybuilder.org/stable/build_tips/#Using-GCC-on-macOS-and-FreeBSD
```suggestion
    CMAKE_TARGET_TOOLCHAIN=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake
```
Side note, it's a bit sad to switch to gcc on BSD systems especially for C++ libraries, as GCC has a different runtime than llvm, and compatibility with other libraries is untested at best (but likely to be highly problematic)

Reply:
Ok I will try again with clang on apple, those warnings in the build with gcc also didn't look like it would work reliably.